flink-ml-iteration/src/test/java/org/apache/flink/iteration/operator/HeadOperatorTest.java

```diff
@@ -318,6 +322,10 @@ public void testPostponeGloballyAlignedEventsAfterSnapshot() throws Exception {
                                                     OperatorUtils.getUniqueSenderId(operatorId, 0)),
                                             0)),
                             new ArrayList<>(harness.getOutput()));
+                    // TODO There might be a potential bug makes the testing hang, which could be
```

Comment:
The root cause is that `RecordingHeadOperatorFactory.latestHeadOperator` is not properly closed. The issue can be fixed by having the following code in `HeadOperatorTest#createHarnessAndRun`:

```
try {
    return runnable.apply(harness);
} finally {
    RecordingHeadOperatorFactory.latestHeadOperator.close();
}
```